api/v1/broker.go

```diff
@@ -1,35 +1,171 @@
 package api
 
 import (
+	"encoding/json"
+	"fmt"
 	"net/http"
+	"strconv"
 
+	"github.com/data-preservation-programs/spade-tenant/db"
+	"github.com/jackc/pgtype"
 	"github.com/labstack/echo/v4"
+	"github.com/labstack/gommon/log"
+	"gorm.io/gorm"
+	"gorm.io/gorm/clause"
 )
 
 func (a *apiV1) ConfigureBrokerRouter(e *echo.Group) {
-	g := e.Group("/broker")
-	g.GET("", a.handleGetTenantsInformation)
-	g.POST("", a.handlePostNotifyTenantService)
+	e.GET("/tenant-state", a.handleGetTenantsState)
+	e.POST("/subscription-event", a.handlePostBrokerSubscriptionEvent)
 }
 
-// handleGetTenantsInformation godoc
-//
-//	@Summary		List of all tenants in a JSON object to be consumed by the broker.
-//	@Security apiKey
-//	@Produce		json
-//	@Success		200	{object}	ResponseEnvelope{response=LabelsResponse}
-//	@Router			/broker [get]
-func (a *apiV1) handleGetTenantsInformation(c echo.Context) error {
-	return c.JSON(http.StatusNotImplemented, CreateErrorResponseEnvelope(c, http.StatusNotImplemented, ""))
+type BrokerResponse []TenantBrokerPayload
+
+type TenantBrokerPayload struct {
+	TenantID                 db.ID             `json:"tenant_id"`
+	TenantStorageContractCID string            `json:"tenant_storage_contract_cid"`
+	TenantSettings           db.TenantSettings `json:"tenant_settings"`
+	TenantAddresses          []db.Address      `json:"tenant_addresses"`
+	CandidateSPs             []CandidateSP     `json:"candidate_sps"`
+	Collections              []db.Collection
+}
+
+type CandidateSP struct {
+	SPID                    db.ID             `json:"sp_id"`
+	ProviderTenantState     string            `json:"provider_tenant_state"`
+	ProviderTenantStateInfo string            `json:"provider_tenant_state_info"`
+	AttributeValues         map[string]string `json:"attribute_values"`
+	ProviderMetadata        pgtype.JSONB      `json:"provider_metadata"` // TODO: struct for what we expect here (i.e, max_bytes_in_flight)
 }
 
-// handlePostNotifyTenantService godoc
-//
-//	@Summary		Allows the broker to notify the tenant service.
-//	@Security apiKey
-//	@Produce		json
-//	@Success		200	{object}	ResponseEnvelope{response=LabelsResponse}
-//	@Router			/broker [post]
-func (a *apiV1) handlePostNotifyTenantService(c echo.Context) error {
-	return c.JSON(http.StatusNotImplemented, CreateErrorResponseEnvelope(c, http.StatusNotImplemented, ""))
+// /broker/tenant-state [get]
```

Comment:
Shouldnt this api get the same love as the other ones with comments

Reply:
Ok, I added some comments to each route but excluded from Swagger by not specifying the `@Router` for them